config/settings.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const Joi = require('joi');
+const Job = require('./job');
+
+const SCHEMA_DISPLAY_JOB_NAME_LENGTH = Joi.number().integer()
+    .min(20)
+    .max(99)
+    .default(20)
+    .description('Job name length to display on workflowGraph')
+    .example(20);
+const SCHEMA_METRICS_DOWNTIME_JOBS = Joi.array().items(
+    Job.jobName
+).description('Jobs to watch for downtime');
+
+const SCHEMA_PIPELINE_SETTINGS = Joi.object()
+    .keys({
+        metricsDowntimeJobs: SCHEMA_METRICS_DOWNTIME_JOBS
```

Comment:
should this be a generic object to store any sort of key-value setting ?

Reply:
sure
I can make it generic if you want to have no validation on `settings`

Reply:
ok, dev needs to be aware of this validation requirement if adding a new setting. I guess this is fine, even if we relax this later, there shouldn't be any schema changes right ?

Reply:
no I don't think so, since we wouldn't be adding any columns